openapi_python_client/cli.py

```diff
@@ -116,6 +116,7 @@ def generate(
     url: Optional[str] = typer.Option(None, help="A URL to read the JSON from"),
     path: Optional[pathlib.Path] = typer.Option(None, help="A path to the JSON file"),
     custom_template_path: Optional[pathlib.Path] = typer.Option(None, **custom_template_path_options),  # type: ignore
+    encoding: Optional[str] = typer.Option(None, help="Set file encoding for client files"),  # type: ignore
```

Comment:
Using an enum of choices would be better than an arbitrary string here... although unless someone has created an enum with every encoding we'll have to manually add supported encoding options. Not the end of the world IMO + gives us a bit more control over what we officially support.